File: Protogen/loxim_protocol/cpp/protocol/packages/V_sc_sendvaluePackage.cpp
#include <string.h>
#include <stdlib.h>
#include "V_sc_sendvaluePackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

V_sc_sendvaluePackage::V_sc_sendvaluePackage(\
		int64_t a_value_id,\
		struct Send_value_flagsMap a_flags,\
		int64_t a_value_type,\
		Package* a_data
		)
{
	value_id=a_value_id;
	flags=a_flags;
	value_type=a_value_type;
	data=a_data;
}

V_sc_sendvaluePackage::V_sc_sendvaluePackage()
{
	value_id=0;
	memset(&flags,0,sizeof(flags));
	value_type=0;
	data=NULL;
}

V_sc_sendvaluePackage::~V_sc_sendvaluePackage()
{
	if(data) delete data;
	data=NULL;
}

void V_sc_sendvaluePackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	writer->writeVaruint(value_id);
	uint8_t tmp_flags;
	Send_value_flagsFactory::getValueByEnumMap(flags, tmp_flags);
	writer->writeUint8(tmp_flags);
	writer->writeVaruint(value_type);
	data->serializeContent(writer);
}

bool V_sc_sendvaluePackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	{bool b; value_id=reader->readVaruint(b);}
	Send_value_flagsFactory::getEnumMapByValue(reader->readUint8(),flags);
	{bool b; value_type=reader->readVaruint(b);}
	data=DataPackagesFactory::createPackageContent(value_type,reader);
	return true;
}

bool V_sc_sendvaluePackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	V_sc_sendvaluePackage* p=(V_sc_sendvaluePackage*)_package;
	if (!(getValue_id()==p->getValue_id())) return false;
	if (!(Send_value_flagsFactory::mapEquals(getFlags(),p->getFlags()))) return false;
	if (!(getValue_type()==p->getValue_type())) return false;
	if (!((!getData() && ! p->getData())||(getData() && p->getData() && (getData()->equals(p->getData()))))) return false;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages/V_sc_sendvaluePackage.java
package pl.edu.mimuw.loxim.protocol.packages;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class V_sc_sendvaluePackage extends Package {
		public final static short ID=(short)33;
	
    	/**
**/
    	private Long value_id=new Long(0);
    	/**
**/
    	private EnumSet<Send_value_flagsEnum> flags=EnumSet.noneOf(Send_value_flagsEnum.class);
    	/**
**/
    	private Long value_type=new Long(0);
    	/**
**/
    	private Package data=null;
    	
	public V_sc_sendvaluePackage()
	{
	}

		public V_sc_sendvaluePackage(
		    	Long a_value_id,
	        	EnumSet<Send_value_flagsEnum> a_flags,
	        	Long a_value_type,
	        	Package a_data    	)
	{
        	value_id=a_value_id;
        	flags=a_flags;
        	value_type=a_value_type;
        	data=a_data;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			value_id=reader.readVaruint();
					short tmp_flags=reader.readUint8();
flags=Send_value_flagsEnum.createEnumMapByValue(tmp_flags);
					value_type=reader.readVaruint();
					data=pl.edu.mimuw.loxim.protocol.packages_data.DataPackagesFactory.getInstance().createPackage(value_type);
data.deserializeContent(reader);
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeVaruint(value_id);		
					writer.writeUint8(Send_value_flagsEnum.toEnumMapValue(flags));		
					writer.writeVaruint(value_type);		
					data.serializeContent(writer);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		V_sc_sendvaluePackage p=(V_sc_sendvaluePackage)obj;
	    if (!(((value_id==p.getValue_id())||((value_id!=null)&&(p.getValue_id()!=null)&&((value_id).equals(p.getValue_id())))))) 
return false;
		if (!(((flags==p.getFlags())||((flags!=null)&&(p.getFlags()!=null)&&((flags).equals(p.getFlags())))))) 
return false;
		if (!(((value_type==p.getValue_type())||((value_type!=null)&&(p.getValue_type()!=null)&&((value_type).equals(p.getValue_type())))))) 
return false;
		if (!(((data==p.getData())||((data!=null)&&(p.getData()!=null)&&((data).equals(p.getData())))))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public Long getValue_id(){return value_id;};
		
    	public void setValue_id(Long a_value_id){ value_id=a_value_id;};
		
        	public EnumSet<Send_value_flagsEnum> getFlags(){return flags;};
		
    	public void setFlags(EnumSet<Send_value_flagsEnum> a_flags){ flags=a_flags;};
		
        	public Long getValue_type(){return value_type;};
		
    	public void setValue_type(Long a_value_type){ value_type=a_value_type;};
		
        	public Package getData(){return data;};
		
    	public void setData(Package a_data){ data=a_data;};
		
    
	 
}


